### DISCUSSION TITLE
Language support for UI elements.

### DISCUSSION THREAD

**Discussion by aliasion** [GitbookIO/community#291] [Feature Requests]
While the existing functionality to publish multiple versions and to handle translations via GitHub actions is fine, it has room for enhancement, especially in terms of user experience (UX) and search engine optimization (SEO).

**Current Limitations:**

- Limited Language Support: The platform currently supports only a few languages. This limitation negatively impacts both the UX and SEO.
- Incorrect HTML Tags: For instance, if content is in Italian, the tag remains `<html lang="en">`, which can detrimentally affect SEO.

**Proposed Feature:**
Custom Translation for UI Elements
Allow users to provide custom translations for specific UI elements (basically a custom map for translation keys).
Since the number of UI elements that require translation isn't extensive, this feature would be manageable for end-users to implement.

**The Impact:**

- SEO: Custom translations would ensure the correct language meta tag is used, optimizing for search engines and enhancing content discoverability.
- UX: Providing users with content in their preferred language improves the overall user experience and can lead to increased engagement.

**Comment by say-borusan** [comment]
This feature will be very useful. I can't use Gibook for documents that I want to create in my local language. When the content is in a different language and the interface components are in a different language, it looks amateurish. 

Also, I think the views on SEO are quite justified.

There should be a simple interface where we can localize the text of the interface components. At least I think it would not be difficult to quickly add Latin languages to Gitbook in line with the requests.

**Comment by MadeByPP** [comment]
I hope this support is added soon. The documentation I wrote is written in a language not supported by the GitBook UI, and the UI is not localized, so it may look amateurish, as @say-borusan 's commented.

There is also the issue of SEO, as mentioned in the text. My document is lagging behind in search results!
